x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo } from 'react';
+import {
+  Criteria,
+  EuiLink,
+  EuiSwitch,
+  EuiTableFieldDataColumnType,
+  EuiBasicTable,
+  EuiBasicTableProps,
+} from '@elastic/eui';
+// import moment from 'moment';
+import type { RulesState } from './rules_container';
+import * as TEST_SUBJECTS from './test_subjects';
+import * as TEXT from './translations';
+import type { RuleSavedObject } from './use_csp_rules';
+
+type RulesTableProps = Pick<
+  RulesState,
+  'loading' | 'error' | 'rules_page' | 'total' | 'perPage' | 'page'
+> & {
+  toggleRule(rule: RuleSavedObject): void;
+  setSelectedRules(rules: RuleSavedObject[]): void;
+  setPagination(pagination: Pick<RulesState, 'perPage' | 'page'>): void;
+  // ForwardRef makes this ref not available in parent callbacks
+  tableRef: React.RefObject<EuiBasicTable<any>>;
+};
+
+export const RulesTable = ({
+  toggleRule,
+  setSelectedRules,
+  setPagination,
+  perPage: pageSize,
+  rules_page: items,
+  page,
+  tableRef,
+  total,
+  loading,
+  error,
+}: RulesTableProps) => {
+  const columns = useMemo(() => getColumns({ toggleRule }), [toggleRule]);
+
+  const euiPagination: EuiBasicTableProps<RuleSavedObject>['pagination'] = {
+    pageIndex: page,
+    pageSize,
+    totalItemCount: total,
+    pageSizeOptions: [1, 5, 10, 25],
+    hidePerPageOptions: false,
+  };
+
+  const selection: EuiBasicTableProps<RuleSavedObject>['selection'] = {
+    selectable: () => true,
+    onSelectionChange: setSelectedRules,
+  };
+
+  const onTableChange = ({ page: pagination }: Criteria<RuleSavedObject>) => {
+    if (!pagination) return;
+    setPagination({ page: pagination.index, perPage: pagination.size });
+  };
+
+  return (
+    <EuiBasicTable
+      ref={tableRef}
+      data-test-subj={TEST_SUBJECTS.CSP_RULES_TABLE}
+      loading={loading}
+      error={error}
+      items={items}
+      columns={columns}
+      pagination={euiPagination}
+      onChange={onTableChange}
+      isSelectable={true}
+      selection={selection}
+      itemId={(v) => v.id}
+    />
+  );
+};
+
+const ruleNameRenderer = (name: string) => (
+  <EuiLink className="eui-textTruncate" title={name}>
+    {name}
+  </EuiLink>
+);
+
+// const timestampRenderer = (timestamp: string) =>
+//   moment.duration(moment().diff(timestamp)).humanize();
+
+interface GetColumnProps {
+  toggleRule: (rule: RuleSavedObject) => void;
+}
+
+const createRuleEnabledSwitchRenderer =
+  ({ toggleRule }: GetColumnProps) =>
+  (value: any, rule: RuleSavedObject) =>
```

Comment:
Please avoid using `any` - this is a boolean (type comes from `attributes.enabled`)